scd/control.c

```diff
@@ -736,6 +764,23 @@ scd_control_cb_accept(int fd, unsigned events, event_io_t *io, void *data)
 	event_add_io(event);
 }
 
+ssize_t
+scd_control_send_event(scd_event_t event, const char *token_uuid)
+{
+	IF_TRUE_RETVAL(event_fd < 0, -1);
+	IF_TRUE_RETVAL(control_event_to_proto(event) == TOKEN_TO_DAEMON__CODE__CMD_UNKNOWN, -1);
+
+	TokenToDaemon msg = TOKEN_TO_DAEMON__INIT;
+	msg.code = control_event_to_proto(event);
+	if (token_uuid)
+		msg.token_uuid = mem_strdup(token_uuid);
+
+	int ret = protobuf_send_message(event_fd, (ProtobufCMessage *)&msg);
+
+	mem_free(msg.token_uuid);
```

Comment:
not allocated if token_uuid == NULL. Perhaps "if (msg.token_uuid)"?

Reply:
done, thanks for spotting